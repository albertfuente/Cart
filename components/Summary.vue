<template>
  <div class="summary-component">
    <div class="summary-component-wrapper">
      <div class="summary-component-container">
        <h3>Resumen del pedido</h3>
        <div class="summary-component-summary-container">
          <div>Total productos IVA Incluido</div>
          <div>{{ items.total_price + items.currency_code }}</div>
        </div>
        <div class="summary-component-summary-container">
          <div>Total de envio (imp.excl)</div>
          <div>
            {{
              items.estimated_shipping_costs === "0.00"
                ? "Envio gratuito"
                : items.estimated_shipping_costs
            }}
          </div>
        </div>
        <div class="summary-component-summary-container">
          <div>Total sin IVA:</div>
          <div>
            {{
              items.estimated_shipping_costs_without_tax + items.currency_code
            }}
          </div>
        </div>
        <div class="summary-component-summary-container">
          <div>Total de impuestos:</div>
          <div>{{ items.total_vat_taxes + items.currency_code }}</div>
        </div>
        <div class="summary-component-summary-container">
          <div class="divider"><b>Total a pagar</b></div>
          <div>
            <b>{{ items.total_price + items.currency_code }}</b>
          </div>
        </div>
      </div>

      <div class="summary-component-container">
        <div class="summary-component-summary-container">
          <form action="/action_page.php">
            <div class="summary-component-account-container">
              <label>Algun descuento? </label><br />
              <input v-model="discount" />
            </div>
          </form>
        </div>
      </div>
    </div>
    <div class="summary-component-container">
      <h3>Crea tu cuenta</h3>

      <form action="/action_page.php">
        <div class="summary-component-account-container">
          <label>Nombre:</label><br />
          <input v-model="name" />
        </div>
        <div class="summary-component-account-container">
          <label>Apellido:</label><br />
          <input v-model="firstName" />
        </div>
        <div class="summary-component-account-container">
          <label>Correo electronico:</label><br />
          <input v-model="email" />
        </div>
        <div class="summary-component-account-container">
          <label>Contrasena:</label><br />
          <input v-model="password" />
        </div>
      </form>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      name: "",
      firstName: "",
      email: "",
      password: "",
      discount: "",
      items: [],
    };
  },
  methods: {
    async fetch() {
      const result = await this.$axios.get(
        `http://demo0915487.mockable.io/cart`
      );
      this.items = result.data.data.attributes;
    },
  },
  mounted() {
    this.fetch();
  },
};
</script>
<style >
.summary-component {
  display: flex;
  justify-content: space-between;
}
.summary-component-container {
  border: 1px solid gray;
  width: 450px;
  margin: 10px;
  padding: 10px;
}
.summary-component-summary-container {
  display: flex;
  justify-content: space-between;
  margin-top: 5px;
}
div.summary-component-summary-container:nth-child(6) {
  border-top: 1px solid black;
  margin-top: 10px;
}
.summary-component-account-container {
  display: flex;
  justify-content: space-between;
  margin-bottom: 5px;
  flex-direction: column;
  margin: 10px 0px;
}
</style>