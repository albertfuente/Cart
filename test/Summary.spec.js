import { shallowMount, mount } from '@vue/test-utils'
import Summary from './../components/Summary.vue'

describe('Summary.vue/Summary', () => {
  it('mounts the component and creates a snapshot inside the __snapshots__ folder', () => {
    const wrapper = shallowMount(Summary)
    expect(wrapper.html()).toMatchSnapshot()
  })


  it('does not display the Summary if items are empty ', () => {
    const items = []
    const wrapper = mount(Summary, { propsData: { items } })

    expect(wrapper.find('.summary-component-items-container').exists()).toBe(false)
  })
})
