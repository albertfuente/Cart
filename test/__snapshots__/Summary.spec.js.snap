// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary.vue/Summary mounts the component and creates a snapshot inside the __snapshots__ folder 1`] = `
"<div class=\\"summary-component\\">
  <div class=\\"summary-component-wrapper\\">
    <div class=\\"summary-component-container\\">
      <h3>Resumen del pedido</h3>
      <div class=\\"summary-component-summary-container\\">
        <div>Total productos IVA Incluido</div>
        <div>NaN</div>
      </div>
      <div class=\\"summary-component-summary-container\\">
        <div>Total de envio (imp.excl)</div>
        <div>

        </div>
      </div>
      <div class=\\"summary-component-summary-container\\">
        <div>Total sin IVA:</div>
        <div>
          NaN
        </div>
      </div>
      <div class=\\"summary-component-summary-container\\">
        <div>Total de impuestos:</div>
        <div>NaN</div>
      </div>
      <div class=\\"summary-component-summary-container\\">
        <div class=\\"divider\\"><b>Total a pagar</b></div>
        <div><b>NaN</b></div>
      </div>
    </div>
    <div class=\\"summary-component-container\\">
      <div class=\\"summary-component-summary-container\\">
        <form action=\\"/action_page.php\\">
          <div class=\\"summary-component-account-container\\"><label>Algun descuento? </label><br> <input></div>
        </form>
      </div>
    </div>
  </div>
  <div class=\\"summary-component-container\\">
    <h3>Crea tu cuenta</h3>
    <form action=\\"/action_page.php\\">
      <div class=\\"summary-component-account-container\\"><label>Nombre:</label><br> <input></div>
      <div class=\\"summary-component-account-container\\"><label>Apellido:</label><br> <input></div>
      <div class=\\"summary-component-account-container\\"><label>Correo electronico:</label><br> <input></div>
      <div class=\\"summary-component-account-container\\"><label>Contrasena:</label><br> <input></div>
    </form>
  </div>
</div>"
`;
